### PR TITLE
Bring StorageObjectInUseProtection feature to GA

### PR DESCRIPTION
**Pull request by pospispa**
**What this PR does / why we need it**:
It brings `StorageObjectInUseProtection` feature to GA.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes N/A

**Special notes for your reviewer**:

**Release note**:

```release-note
StorageObjectInUseProtection feature is GA.
```

### CODE REVIEW

**Comment by pospispa** [comment]
/sig storage
/kind feature
@jsafrane @NickrenREN @msau42 PTAL

File: cmd/kube-controller-manager/app/core.go [hunk 1]
```diff
@@ -395,24 +395,18 @@ func startGarbageCollectorController(ctx ControllerContext) (bool, error) {
 }
 
 func startPVCProtectionController(ctx ControllerContext) (bool, error) {
-	if utilfeature.DefaultFeatureGate.Enabled(features.StorageObjectInUseProtection) {
```
Comment (liggitt) [inline cmd/kube-controller-manager/app/core.go]: We don't typically remove the gate to go GA, we flip the feature status, and that's it.

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -201,12 +201,6 @@ const (
 	// Enable Block volume support in containers.
 	BlockVolume utilfeature.Feature = "BlockVolume"
 
-	// owner: @pospispa
-	// beta: v1.10
-	//
-	// Postpone deletion of a PV or a PVC when they are being used
-	StorageObjectInUseProtection utilfeature.Feature = "StorageObjectInUseProtection"
```
Comment (liggitt) [inline pkg/features/kube_features.go]: Removal of a gate breaks startups that explicitly enabled it. I'd expect this to remain, with GA status, either indefinitely, or for an announced deprecation period

**Comment by NickrenREN** [comment]
PV/PVC protection controllers now share much similar code , I want to combine them as well as e2e tests, like admission plugin does, and simplify the code. Could we postpone this promotion and do it in 1.12 or after the simplification during 1.11 ?
@pospispa @jsafrane

**Comment by pospispa** [comment]
@NickrenREN mentioned that it'd be better to firstly combine code of the `pvc-protection-controller` and `pv-protection-controller` and afterwards bring the `StorageObjectInUseProtection` feature to GA.
I agree with that so I'm closing this PR.

**Comment by NickrenREN** [comment]
I will send out a PR for that

**Comment by NickrenREN** [comment]
PR: https://github.com/kubernetes/kubernetes/pull/61385

**Pull request closed without merging**